qutip/solver/multitraj.py

```diff
@@ -0,0 +1,494 @@
+            into which add more trajectories. The result will be changed
+            inplace.
+
+        ntraj : int
+            Number of trajectories to add.
+
+        timeout : float, optional
+            Maximum time in seconds for the trajectories to run. Once this time
+            is reached, the simulation will end even if the number
+            of trajectories is less than ``ntraj``. The map function, set in
+            options, can interupt the running trajectory or wait for it to
+            finish. Set to ``0`` to disable
+
+        target_tol : {float, tuple, list}, optional
+            If a float, it is read as absolute tolerance.
+            If a pair of float: absolute and relative tolerance in that order.
+            Lastly, target_tol can be a list of pairs of (atol, rtol) for each
+            e_ops.
+
+        seed : {int, list(int)} optional
+            Seed or list of seeds for each trajectories.
+
+        Return
+        ------
+        results : :class:`qutip.solver.MultiTrajResult`
+            Results of the evolution. States and/or expect will be saved. You
+            can control the saved data in the options.
+        """
+        start_time = time()
+        if result.solver_id != id(self):
+            raise ValueError("Result class was not created by this solver")
+        seeds = self._read_seed(seed, ntraj)
+        map_func = get_map[self.options.mcsolve['map']]
+        map_kw = {
+            'timeout': timeout or self.options.mcsolve['timeout'],
+            'job_timeout': self.options.mcsolve['job_timeout'],
+            'num_cpus': self.options.mcsolve['num_cpus'],
+        }
+        result._target_ntraj = ntraj + result.num_traj
+
+        if target_tol:
+            result.set_expect_tol(target_tol)
+        map_func(
+            self._traj_solver._run, seeds,
+            (result.initial_state, result.tlist, result.e_ops),
+            reduce_func=result.add, map_kw=map_kw,
+            progress_bar=self.options["progress_bar"],
+            progress_bar_kwargs=self.options["progress_kwargs"]
+        )
+        result.stats['run time'] += time() - start_time
+        result.stats.update(self.stats)
+        return result
+
+    @property
+    def options(self):
+        return self._options
+
+    @options.setter
+    def options(self, new):
+        if new is None:
+            new = self.optionsclass()
+        elif isinstance(new, dict):
+            new = self.optionsclass(**new)
+        elif isinstance(new, self.optionsclass):
+            new = copy(new)
+        else:
+            raise TypeError("options must be an instance of" +
+                            str(self.optionsclass))
+        self._options = new
+
+    def _argument(self, args):
+        """Update the args, for the `rhs` and `c_ops` and other operators."""
+        self.rhs.arguments(args)
+
+    @property
+    def traj_args(self):
+        return (self.rhs,)
```

Comment:
Should this not be overwritten by McSolver to include c_ops?

Reply:
It is.